### ISSUE TITLE
TextInput doesn't lose focus after dismissing keyboard on some Android devices

### ISSUE THREAD

**Issue by Ol1va** [facebook/react-native#33532]
### Description

**Current Behavior**

When TextInput is focused and keyboard is shown, pressing hardware back button dismisses the keyboard but doesnt't blur the TextInput. Pressing on the TextInput again doesn't trigger the keyboard.

I noticed it happening on these devices: Mobistart V1 (android v 9), Meizu M3s (android v 5.1), Honor 8 (android v 7).
This behaviour doesn't seem to be dependant on Android version or brand.
Also this behaviour appears in every React Native app.

**Expected Behavior**

The TextInput should lose focus after hardware back button press.

https://user-images.githubusercontent.com/43006289/161037494-008f1c5c-bc35-4eb4-a633-47dc3211ad80.mp4

### Version

0.65.1

### Output of `npx react-native info`

No warnings or errors apper in console.

### Steps to reproduce

Basically it happens with any TextInput.

`<TextInput />`

**Comment by chrisbobbe** [comment]
Probably related (the issue is closed but still reproduces): https://github.com/facebook/react-native/issues/19366

**Comment by Ol1va** [comment]
> 

That solution doens't work for me, because it suggests to make handler for each TextInput in every React Naive project.

**Comment by chrisbobbe** [comment]
Hmm, yeah, that's unfortunate. I'm hoping the link might be helpful for you or anyone else who has time to dig into React Native's internals to find the bug and propose a fix. I'll post here if I manage to do that too. 🙂

**Comment by ghiculescualexandru** [comment]
Any updates on this?

**Comment by Ol1va** [comment]
> Any updates on this?

Unfortunately, no

**Comment by ghiculescualexandru** [comment]
> > Any updates on this?
> 
> Unfortunately, no

I managed to work around by adding a ref to the text input and a keyboard listener to blur the text input when the keyboard hide. 
Here is the logic:


```
  const localInputRef = useRef<TextInput>();
  
  const keyboardDidHideCallback = () => {
     localInputRef.current.blur?.();   
  }

  useEffect(() => {
     const keyboardDidHideSubscription = Keyboard.addListener('keyboardDidHide', keyboardDidHideCallback);

     return () => {
      keyboardDidHideSubscription?.remove();
    };
  }, []);

 <TextInput
            ref={(ref) => {
              localInputRef && (localInputRef.current = ref as any);
            }}
 .....
 </TextInput>
```

**Comment by sankparatkar** [comment]
@ghiculescualexandru you did this for every TextInput box?

**Comment by ghiculescualexandru** [comment]
Only needed it in a single place, however, you can create a component with this behaviour and use it everywhere

**Comment by jwallet** [comment]
this happening in "react-native": "0.71.3"

**Comment by AlkanV** [comment]
we also experience same issue on Android, When user press native back button to close keyboard, textinput of react native stays focused.

**Comment by alainib** [comment]
yep still have this issue. any fix without having to add ref for each input ?

**Comment by firstpersoncode** [comment]
> This is still an issue on react native v0.69.6, unsure if fixed in .7

still happening on .7x

**Comment by fabriziobertoglio1987** [comment]
Do you still experience this issue? 

I have four years of experience maintaining facebook/react-native and I specialize in the Text and TextInput components. I currently have 58 [facebook/react-native PRs](https://github.com/facebook/react-native/pulls?q=is%3Apr+author%3Afabriziobertoglio1987+). 

If you still experience this issue, I will prepare a patched release with the fix.

Thanks a lot

**Comment by shunnobir** [comment]
> Do you still experience this issue?
> 
> I have four years of experience maintaining facebook/react-native and I specialize in the Text and TextInput components. I currently have 58 [facebook/react-native PRs](https://github.com/facebook/react-native/pulls?q=is%3Apr+author%3Afabriziobertoglio1987+).
> 
> If you still experience this issue, I will prepare a patched release with the fix.
> 
> Thanks a lot

I am experiencing the same issue.

**Comment by simonadenic** [comment]
Still an issue on 0.72.7

**Comment by kamajus3** [comment]
Nothing new on 0.74.3

**Comment by pranshuchittora** [comment]
😢 still broken on 0.75.3

**Comment by ashfaqdevdksa** [comment]
@cipolleschi @cortinico Please fix this.

**Comment by cipolleschi** [comment]
I just test this in a recent version (0.77.0-rc.3 and 0.76.5) of React Native and it seems to work fine.
I can:
- focus on the Text Field, the keyboard appear.
- press the back button, the keyboard disappear, the text field remains focused.
- tapping on the Text Field again make the keyboard appear again.\

| 0.77.0-rc.3 | 0.76.5 |
| --- | --- |
| <img src="https://github.com/user-attachments/assets/3253ee81-41df-4916-9520-6bb65da0031d" height=480 /> | <img src="https://github.com/user-attachments/assets/6e30c6f2-7b0c-4391-bfd8-a5193c989b3e" height=480 /> |

I think this has been fixed. Can someone else test this out?

**Comment by samcodefish** [comment]
> I just test this in a recent version (0.77.0-rc.3 and 0.76.5) of React Native and it seems to work fine. I can:
> 
> * focus on the Text Field, the keyboard appear.
> * press the back button, the keyboard disappear, the text field remains focused.
> * tapping on the Text Field again make the keyboard appear again.\
> 
> I think this has been fixed. Can someone else test this out?

Hi @cipolleschi , I think maybe you have misunderstood the issue?

The problem is that the TextInput **remains focussed** when the keyboard is dismissed. Which your demo illustrates.

The desired behaviour would be for the TextInput to **become unfocussed** when the keyboard is dismissed with the back button press.

**Comment by alanggdev** [comment]
Any updates on this?

**Comment by sayan-avantai** [comment]
any update on this?

**Comment by jdziadek** [comment]
Bump

**Comment by Bowuigi** [comment]
Can reproduce in RN 0.81.4, Expo 54.0.12, Android 11. Every workaround mentioned (here and in #19366, namely, blur-then-focus and `Keyboard.dismiss()`) works though.

**Comment by ramchandra-st** [comment]
Bump

**Comment by BejPastan** [comment]
Still doesn't work
